pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -318,12 +340,33 @@ func buildService(mdb mdbv1.MongoDB) corev1.Service {
 		Build()
 }
 
+func getCurrentAutomationConfig(client mdbClient.Client, mdb mdbv1.MongoDB) (automationconfig.AutomationConfig, error) {
+	currentCm := corev1.ConfigMap{}
+	currentAc := automationconfig.AutomationConfig{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: mdb.ConfigMapName(), Namespace: mdb.Namespace}, &currentCm); err != nil {
+		if !errors.IsNotFound(err) {
+			return automationconfig.AutomationConfig{}, err
+		}
```

Comment:
so we discard is not found errors?

I find this code convoluted: can you please make it simpler?

Reply:
I think making removing the 1st if would make this clearer

Reply:
right now I think that if you get a is not found error while fetching the config map, you will return an empty automation config with no error
if this is intended: make it explicit and remove the else part

Reply:
Yes @LouisPlisso you are right we are returning an empty AC with no error in this case.

A more explicit way to express it would look like this:

```go
	if err := client.Get(context.TODO(), types.NamespacedName{Name: mdb.ConfigMapName(), Namespace: mdb.Namespace}, &currentCm); err != nil {
		if errors.IsNotFound(err) {
			return automationconfig.AutomationConfig{}, nil
		} else {
			return automationconfig.AutomationConfig{}, err
		}

	}
	if err := json.Unmarshal([]byte(currentCm.Data[AutomationConfigKey]), &currentAc); err != nil {
		return automationconfig.AutomationConfig{}, err
	}
	return currentAc, nil
```

would you consider this (with appropriate comments) a better solution?

Reply:
way easier to read: I'd just remove the else (just return unindented as the if is returning), but it's whatever you prefer

Reply:
or maybe if errors.IsNotFound { err = nil } and then return AutomationConfig{}, err unconditionnaly

maybe that's the clearest?

Reply:
and this one only needs comment on why we hide the not found error, everything else is self explanatory

Reply:
I like the idea of having `if errors.isNotFound {err = nil }`

 I think it might be even taken a step further and have something analogous to what we have in [k8s_condition.py](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/scripts/dev/k8s_conditions.py#L48): we could have a function which takes a number of statuses (in this case it would be _metav1.StatusReasonNotFound_) and returns `nil` if the error is one of the provided ones.

So that would look like in our case:
```go
if err := client.Get(context.TODO(), types.NamespacedName{Name: mdb.ConfigMapName(), Namespace: mdb.Namespace}, &currentCm); err != nil {
         err  = makeErrorNilIfStatusIs(err, metav1.StatusReasonNotFound)
	return automationconfig.AutomationConfig{}, err
}
```

This is probably outside the scope of this PR though

Reply:
this `makeErrorNilIfStatusIs` functionality already exists, in the form of `client.IgnoreNotFound(err)` (`"sigs.k8s.io/controller-runtime/pkg/client"`)

Reply:
nice so in the end we could have

```
if err := xxx; err != nil {
    return automationconfig.AutomationConfig{}, client.IgnoreNotFound(err)
}
```
no more else

so much clearer